portable/NetworkInterface/RISC-V/riscv_hal_eth.c

```diff
@@ -479,17 +479,29 @@ int DmaSetup(XAxiDma *DmaInstancePtr, u16 AxiDmaDeviceId)
 
         #ifdef __CHERI_PURE_CAPABILITY__
             RxFrameBufRef = cheri_build_data_cap(RX_FRAME_BUF_ADDR, sizeof(EthernetFrame)*RXBD_CNT,
+                               __CHERI_CAP_PERMISSION_PERMIT_EXECUTE__ |
                                __CHERI_CAP_PERMISSION_PERMIT_LOAD__ |
                                __CHERI_CAP_PERMISSION_PERMIT_STORE__);
             TxFrameBufRef = cheri_build_data_cap(TX_FRAME_BUF_ADDR, sizeof(EthernetFrame)*TXBD_CNT,
+                               __CHERI_CAP_PERMISSION_PERMIT_EXECUTE__ |
                                __CHERI_CAP_PERMISSION_PERMIT_LOAD__ |
                                __CHERI_CAP_PERMISSION_PERMIT_STORE__);
             RxBdSpaceRef = cheri_build_data_cap(RXBD_SPACE_ADDR, RXBD_SPACE_BYTES,
-                               __CHERI_CAP_PERMISSION_PERMIT_LOAD__ |
-                               __CHERI_CAP_PERMISSION_PERMIT_STORE__);
+                               __CHERI_CAP_PERMISSION_GLOBAL__
+                               | __CHERI_CAP_PERMISSION_PERMIT_EXECUTE__
+                               | __CHERI_CAP_PERMISSION_PERMIT_LOAD__
+                               | __CHERI_CAP_PERMISSION_PERMIT_STORE__
+                               | __CHERI_CAP_PERMISSION_PERMIT_STORE_CAPABILITY__
+                               | __CHERI_CAP_PERMISSION_PERMIT_STORE_LOCAL__
```

Comment:
Why do we need store local permission?

Reply:
I think at one point I was getting exceptions when I was trying to store capabilities in the Buffer Descriptors, and the exception code was a local store code. It's most likely an error on my part; I'll have a look and see what I did wrong.

Reply:
Yeah, you'd likely get that exception if you're trying to save a non-global capability (e.g., one defined within a function) inside the RxFrame buffer itself.